debezium-embedded/src/main/java/io/debezium/embedded/AsyncEngineConfig.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.embedded;
+
+import io.debezium.config.Field;
+
+/**
+ * Configuration options specific to {@link AsyncEmbeddedEngine}.
+ *
+ * @author vjuranek
+ */
+public interface AsyncEngineConfig extends EmbeddedEngineConfig {
+
+    int AVAILABLE_CORES = Runtime.getRuntime().availableProcessors();
+
+    /**
+     * An optional field that specifies the maximum amount of time to wait for task lifecycle operation, i.e. for starting and stopping the task.
+     */
+    Field TASK_MANAGEMENT_TIMEOUT_MS = Field.create("task.management.timeout.ms")
+            .withDescription("Time to wait for task's lifecycle management operations (starting and stopping), given in milliseconds. "
+                    + "Defaults to 5 seconds (5000 ms).")
+            .withDefault(5_000L)
+            .withValidation(Field::isPositiveInteger);
+
+    /**
+     * An optional field that specifies the number of threads to be used for processing CDC records.
+     */
+    Field RECORD_PROCESSING_THREADS = Field.create("record.processing.threads")
```

Comment:
I was not able to find the usage of this parameter (in this commit). Where is it used? Furthermore, depending on the workload and the environment over-committing (at least on bare-metal) can make sense, or limiting it using upper limits (like sum of running/scheduled/max tasks). For example environments like dedicated servers (probably bare-metal) for running the engine vs. 96-core servers running many different workloads in parallel.
What is meant by "processing CDC records"?

Reply:
It's a good point to make this field `String` which would accept some predefined value and eventually parse the string as a number.
However, I'm not sure I understand why to limit number of threads to some predefined value. Is the concern here possible delay due to context switching on overloaded environments?

Reply:
Well, unless we test with such "absurd" HW setup I'd avoid to large default pools. It is more precautionary approach than anything else.